pages/[wiki]/[article]/index.tsx

```diff
@@ -0,0 +1,145 @@
+      .use(rehypeStringify)
+      .processSync(markdown)
+      .toString();
+
+    setMarkdownString(markdownText);
+  }, [markdown]);
+
+  if (router.isFallback) {
+    return <div>Loading...</div>;
+  }
+  return (
+    <>
+      <Head>
+        <title>{seoTitle}</title>
+        {/* @TODO - add description */}
+      </Head>
+      <div className="max-w-7xl m-auto" id="markdown" dangerouslySetInnerHTML={{ __html: markdownString }} />
+    </>
+  );
+};
+
+export const getStaticProps: GetStaticProps = async ({ params, locale }) => {
+  const param = params!.article;
+  const articles = fs.readFileSync(path.join(process.cwd(), 'articles.json'), { encoding: 'utf-8' });
+
+  const currentWikiPage = (JSON.parse(articles) as string[]).find((article: string) => normalizeWikiPagesUrl(article) === param);
+
+  const response = await fetch(`https://raw.githubusercontent.com/wiki/cesko-digital/movapp/${currentWikiPage}.md`);
+  if (response.status === 404) {
+    return {
+      notFound: true,
+    };
+  }
+  const markdown = await response.text();
+
+  return {
+    props: {
+      markdown,
+      title: currentWikiPage?.replace(/-{1,5}/g, ' '),
+      ...(await serverSideTranslations(locale ?? 'cz', ['common'])),
+    },
+    revalidate: 10,
+  };
+};
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  // list of pages which should not be fetched in nested url ex. wiki/...
+  const excludedPages = ['sk', 'cs', 'pl', 'uk-pl', 'uk-cs', 'uk-sk'];
+
+  const mainLanguage = process.env.NEXT_PUBLIC_COUNTRY_VARIANT || 'cs';
+
+  /** Recursively finds all internal links and creates paths for pages which should be generated
+   *
+   * @param articleName - article name to be fetched
+   * @param tracking - tracks which articles should be fetched
+   * @param paths - list of paths from which pages should be generated
+   * @returns
+   */
+  const fetchWikiArticle = async (articleName: string, tracking: string[], paths: string[]): Promise<string[]> => {
+    const response = await fetch(`https://raw.githubusercontent.com/wiki/cesko-digital/movapp/${articleName}.md`);
+    const markdown = await response.text();
+
+    const { links } = markdownLinkExtractor(markdown);
+
+    links.forEach((link: string) => {
+      if (link.includes('https') || tracking.includes(link) || excludedPages.includes(link)) return;
+      paths.push(link);
+      tracking = [...tracking, link];
+    });
+
+    const nextFetchURL = tracking.shift();
+    if (!nextFetchURL) return paths;
+    return await fetchWikiArticle(nextFetchURL, tracking, paths);
+  };
+
+  const mainLanguageArticles = await fetchWikiArticle(mainLanguage, [], []);
+  const ukArticles = await fetchWikiArticle(`uk-${mainLanguage}`, [], []);
+  const otherArticles = await fetchWikiArticle('Home', [], []);
+
+  // since we need to normalize articles' name in order to create paths
+  // and it is not possible to pass any data through getStaticPaths to getStaticProps besides params
```

Comment:
Hmm this seems strange "and it is not possible to pass any data through getStaticPaths to getStaticProps besides params". But isn't the path (as param) all you need? GetStaticPaths should return an array of paths. The path gets passed to the Article component as a parameter. The article component fetches markdown using the path and renders it. Why do you need to store paths in a file?

Reply:
Like look at the first example here https://nextjs.org/docs/routing/dynamic-routes 

Why are we not simply getting the path inside `pages/wiki/[articlePath].tsx` 

like 

```
const WikiArticle = () => {
  const router = useRouter()
  const { articlePath } = router.query

  return <p>Post: {getWikiArticle(articlePath)}</p>
}
```

Reply:
For example we have page with `https://github.com/cesko-digital/movapp/wiki/Ukrajistina-v-Netflixu`. In order to fetch it I need to have access to `Ukrajistina-v-Netflixu` but we don't want to have it in the path with diacritics and capital letter. So we should normalize it for path but in order to fetch it in getStaticProps we need to use initial slug `Ukrajistina-v-Netflixu` which we don't have access to since path is already normalized to `ukrajinstina-v-netflixu`

Reply:
If this is a problem, then we can in wiki try to make every page prenomalized  with lower letters and without diacritics.

It is writter by human, we can make some errors there, but we can correct them later and try to make every wiki page path looks correctly.

Reply:
I can change the path into: https://github.com/cesko-digital/movapp/wiki/ukrajistina-v-netflixu if this helps (and for another wiki pages too)

Reply:
if it is possible it would be better to have it prenormalized

Reply:
Ok, I have just normalized every page in wiki to lowercased letters only and without diacritics. (Even the *Home* page I changed into *home*).

I will do the same for every new wiki page.
It will be good if Movapp do not crash if it founds some page that is not normalized (becouse someone made a mistake). But it is no problem if Movapp ignores that page (its writer will realize this soon and will correct its mistake).

Is this good? Is this enough?

Reply:
yeah it should be enough. Thank you